hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockTest.groovy

```diff
@@ -1,75 +1,74 @@
 package pl.allegro.tech.hermes.mock
 
-import com.fasterxml.jackson.core.JsonProcessingException
-import com.fasterxml.jackson.databind.ObjectMapper
-import org.junit.ClassRule
+import org.junit.Rule
 import pl.allegro.tech.hermes.test.helper.endpoint.HermesPublisher
-import pl.allegro.tech.hermes.test.helper.message.TestMessage
-import spock.lang.Shared
 import spock.lang.Specification
 
 class HermesMockTest extends Specification {
-    @ClassRule
-    @Shared
+    @Rule
     HermesMock hermes = new HermesMock(56789);
 
     protected HermesPublisher publisher = new HermesPublisher("http://localhost:56789");
 
     def "should receive a message"() {
         given:
         def topicName = "my-test-topic"
+        hermes.addTopic(topicName)
 
         when:
         publish(topicName)
 
         then:
-        hermes.assertTopic(topicName)
+        hermes.expectSingleMessageOnTopic(topicName)
     }
 
     def "should receive 3 messages"() {
         given:
-        def topicName = "my-test-topic-3"
+        def topicName = "my-test-topic"
+        hermes.addTopic(topicName)
 
         when:
         3.times {
             publish(topicName)
         }
 
         then:
-        hermes.expectTopic(3, topicName)
+        hermes.expectMessagesOnTopic(3, topicName)
     }
 
-//    def "should receive 2 messages + 1 delayed"() {
-//        given:
-//        def topicName = "my-test-topic-3"
-//
-//        when:
-//        2.times {
-//            publish(topicName)
-//        }
-//
-//        new Thread(new Runnable() {
-//            void run() {
-//                sleep(1000)
-//                publish(topicName)
-//            }
-//        }).start();
-//
-//        then:
-//        hermes.expectTopic(3, topicName)
-//    }
+    def "should receive 2 messages + 1 delayed"() {
+        given:
+        def topicName = "my-test-topic-3"
+        hermes.addTopic(topicName)
+
+        when:
+        2.times {
+            publish(topicName)
+        }
+
+        new Thread(new Runnable() {
+            void run() {
+                sleep(500)
+                publish(topicName)
+            }
+        }).start();
```

Comment:
[There is extension method in Groovy](http://docs.groovy-lang.org/docs/groovy-next/html/groovy-jdk/java/lang/Thread.html#start(groovy.lang.Closure)):
```
Thread.start {
  sleep(500)
  publish(topicName)
}
```